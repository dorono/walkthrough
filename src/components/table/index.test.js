import React from 'react';
import {shallow} from 'enzyme';
import {Table} from './index';

const defaultProps = {
    columns: [
        'col1',
        'col2',
        'col3_short|||col3_long',
    ],
    ellipsis: [1, 2],
    rows: [{
        field1: 'row1_test1',
        field2: 'row1_test2',
        field3: 'row1_test3',
    },
    {
        field1: 'row2_test1',
        field2: 'row2_test2',
        field3: 'row2_test3',
    }],
};

function createTableRows(rowVal, idx) {
    return (
        <tr key={idx}>
            <td>{rowVal.field1}</td>
            <td>{rowVal.field2}</td>
            <td>{rowVal.field3}</td>
        </tr>
    );
}

describe('Table', () => {
    it('should render without errors', () => {
        shallow(
            <Table {...defaultProps}>
                {(row, index) => createTableRows(row, index)}
            </Table>,
        );
    });

    it('should match snapshot with 2 cols set for ellipses', () => {
        expect(shallow(
            <Table {...defaultProps}>
                {(row, index) => createTableRows(row, index)}
            </Table>,
        )).toMatchSnapshot();
    });

    it('should match snapshot with 1 col set for ellipses', () => {
        defaultProps.ellipsis = 2;
        expect(shallow(
            <Table {...defaultProps}>
                {(row, index) => createTableRows(row, index)}
            </Table>,
        )).toMatchSnapshot();
    });
});
