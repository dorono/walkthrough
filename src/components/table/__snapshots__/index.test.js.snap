// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should match snapshot with 1 col set for ellipses 1`] = `
<table
  className="primary interactive"
  onClick={[Function]}
>
  <thead
    className="left"
  >
    <tr>
      <th
        className="header"
        key="col1"
      >
        <span>
          col1
        </span>
      </th>
      <th
        className="header"
        key="col2"
      >
        <span>
          col2
        </span>
      </th>
      <th
        className="header"
        key="col3_short|||col3_long"
      >
        <span>
          <span
            className="long"
          >
            col3_short
          </span>
          <span
            className="short"
          >
            col3_long
          </span>
        </span>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td>
        row1_test1
      </td>
      <td>
        row1_test2
      </td>
      <td
        className="ellipsis"
        key="2"
      >
        row1_test3
      </td>
    </tr>
    <tr
      key="1"
    >
      <td>
        row2_test1
      </td>
      <td>
        row2_test2
      </td>
      <td
        className="ellipsis"
        key="2"
      >
        row2_test3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table should match snapshot with 2 cols set for ellipses 1`] = `
<table
  className="primary interactive"
  onClick={[Function]}
>
  <thead
    className="left"
  >
    <tr>
      <th
        className="header"
        key="col1"
      >
        <span>
          col1
        </span>
      </th>
      <th
        className="header"
        key="col2"
      >
        <span>
          col2
        </span>
      </th>
      <th
        className="header"
        key="col3_short|||col3_long"
      >
        <span>
          <span
            className="long"
          >
            col3_short
          </span>
          <span
            className="short"
          >
            col3_long
          </span>
        </span>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td>
        row1_test1
      </td>
      <td
        className="ellipsis"
        key="1"
      >
        row1_test2
      </td>
      <td
        className="ellipsis"
        key="2"
      >
        row1_test3
      </td>
    </tr>
    <tr
      key="1"
    >
      <td>
        row2_test1
      </td>
      <td
        className="ellipsis"
        key="1"
      >
        row2_test2
      </td>
      <td
        className="ellipsis"
        key="2"
      >
        row2_test3
      </td>
    </tr>
  </tbody>
</table>
`;
